d4/src/ssio/reader.rs

```diff
@@ -131,8 +131,9 @@ impl<R: Read + Seek> D4TrackReader<R> {
             if table_ref.chrom_id == chrom_id {
                 let overlap_begin = table_ref.begin.max(begin);
                 let overlap_end = table_ref.end.min(end);
+
                 if overlap_begin < overlap_end {
-                    if overlap_begin == table_ref.begin || self.sfi.is_none() {
+                    if overlap_begin + 1 == table_ref.begin || self.sfi.is_none() {
```

Comment:
@38 do you think this +1 belongs here? I've forgotten all my previous testing, but I htink this might be required.

Reply:
I don't think we need +1 at this point. Basically what is happening at this point is, we check if we need to read a secondary partition from the middle of the stream. When overlap_begin equals stream's begin, it means we can read the partition from the begin, therefore no need to visit the index. 

I suspect what you previously seen is unrelated to this part, but might be another bug.

Reply:
I'ver reverted all changes in ssio/reader.rs